core/typeSugar.ml

```diff
@@ -1515,6 +1522,20 @@ end
                "but the currently allowed effects are" ^ nli () ^
                 code ppr_lt)
 
+    let escaped_quantifier ~pos ~var ~annotation ~escapees =
+      let escaped_tys = List.map snd escapees in
+      build_tyvar_names (annotation :: escaped_tys);
+      let policy () = { (error_policy ()) with Types.Print.quantifiers = true } in
+      let display_ty (var, ty) =
+        let ppr_ty = Types.string_of_datatype ~policy ty in
```

Comment:
Again, not `Types.string_of_datatype` but `string_of_datatype`. `Types.string_of_datatype` will refresh type variables on each call, `string_of_datatype` will not.

Reply:
Also, might be worth inlining `ppr_ty`.

Reply:
Gotcha -- I think you mean `show_type`? But that doesn't allow me to pass in a policy (and I require one here in order to show the quantifiers in the error message).

I think my best bet would therefore to be explicit and say
```
Types.string_of_datatype ~policy ~refresh_tyvar_names=false
```

Reply:
Why not:

```OCaml
Printf.sprintf "%s: %s" var (string_of_datatype ~policy ty) in
```
Am I missing something obvious here?

Reply:
Ah, yes. I am missing something obvious - just realized what you mean by `show_type.` Give me a moment to look at the code.

Reply:
`string_of_datatype` doesn't seem to exist in the scope unqualified:
```
File "core/typeSugar.ml", line 1530, characters 37-55:
Error: Unbound value string_of_datatype
```

Reply:
Yes, you're absolutely right. Apparently the API is a bit confusing and easily leads to bugs (e.g. on line 1501 in `typeSugar.ml`). This probably needs rethinking.

I wonder of it would make sense to display quantifiers by default in error messages?

Reply:
But either way, this is beyond the scope of this PR.